Generators/src/GeneratorFromFile.cxx

```diff
@@ -55,15 +55,28 @@ void GeneratorFromFile::SetStartEvent(int start)
   }
 }
 
-bool isOnMassShell(TParticle const& p)
+bool GeneratorFromFile::isOnMassShell(TParticle& p)
```

Comment:
since this function is now changing the particle `p`, I would propose to rename it to something like `fixKinematics` or similar.

Reply:
ok, renamed to rejectOrFixKinematics